### ISSUE TITLE
Arch Linux app black

### ISSUE THREAD

**Issue by Karlos5000** [amir1376/ab-download-manager#291]
Arch linux KDE Wayland, manager starts but app stays black

**Comment by YouCanTouCan** [comment]
I have the same issue seemingly on Fedora KDE Wayland - I open the app, and it's just a black rectangle in the center of my screen.

![Screenshot_20241215_190341](https://github.com/user-attachments/assets/fb7fbaa8-e87e-4069-b400-2be6add2ba42)

**Comment by Digital-Yeti** [comment]
Gnome wayland does not even show the black rectangle, but a blank ghost window that cannot be selected. Also happens when building from source.

**Comment by Digital-Yeti** [comment]
Hey @Karlos5000 and @YouCanTouCan I found a solution for this that worked for me with identical issues on the same distro: https://github.com/amir1376/ab-download-manager/issues/94#issuecomment-2406937262
TLDR for anyone who doesnt wanna click around:

> but you can override it by SKIKO_RENDER_API environment variable  
possible values for Windows are  
OPENGL (you are using it now, Open GL)  
DIRECT3D (uses DirectX)  
SOFTWARE_COMPAT  
SOFTWARE_FAST  
For example  
SKIKO_RENDER_API=SOFTWARE_FAST  
After changing your environment variable, make sure the app process is not running and start the app again let see if its  work?

I was able to `export SKIKO_RENDER_API=SOFTWARE_FAST` and run the app! It did have scaling issues but this gets the app to launch at all (still with logging errors in console)
```
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
```

**Comment by YouCanTouCan** [comment]
I can also report that the environmental variable `SKIKO_RENDER_API=SOFTWARE_FAST` mostly works! As reported, the app now opens, but with scaling issues. For the sake of testing, I tried the other options too. With Direct3d, the app wont open, but with Software_Compat or Software_Fast it opens with the scaling issues.

**Comment by Digital-Yeti** [comment]
Hopefully @amir1376 can add this as an automatic fallback scenario, if they cannot for whatever reason resolve this issue to at minimum get it running on Arch.

**Comment by amir1376** [comment]
> Hopefully @amir1376 can add this as an automatic fallback scenario, if they cannot for whatever reason resolve this issue to at minimum get it running on Arch.

Hi there.
This is done by compose framewrok. they already use falback `renderApi` when there is an error when trying to use OpenGL. but in your case it doesn't get any error. everything seems normal but the actual picture output is broken.

This problem is related to skia.( or a missing dependency in your OS. which causes the problem when skia tries to use OpenGL. it should throw some error but it don't )
and this is why when you set `renderApi` to `SOFTWARE_FAST` (cpu) instead of default OPENGL (gpu) it works!

**Comment by ItsYa5een** [comment]
edit the  .desktop file and add
env SKIKO_RENDER_API=SOFTWARE_FAST
right after Exec=
it should look like
Exec=env SKIKO_RENDER_API=SOFTWARE_FAST /home/USER NAME/.local/ABDownloadManager/bin/ABDownloadManager

this resolved the issue for me

**Comment by RoninTurkey** [comment]
I'm on CachyOS and even adding "SKIKO_RENDER_API=SOFTWARE_FAST" to Environment variables doesn't fix the issue.

**Comment by ItsYa5een** [comment]
> I'm on CachyOS and even adding "SKIKO_RENDER_API=SOFTWARE_FAST" to Environment variables doesn't fix the issue.

try diffrent ones
OPENGL
SOFTWARE_COMPAT
SOFTWARE_FAST
example
SKIKO_RENDER_API=OPENGL

**Comment by chethan62** [comment]
> > I'm on CachyOS and even adding "SKIKO_RENDER_API=SOFTWARE_FAST" to Environment variables doesn't fix the issue.
> 
> try diffrent ones OPENGL SOFTWARE_COMPAT SOFTWARE_FAST example SKIKO_RENDER_API=OPENGL
> 
> if this doesnt work make sure you're adding the env variables the right way, following this https://wiki.archlinux.org/title/Environment_variables

iam using cachy os it is working for me   i used this command   use it in terminal fish ,     set -x SKIKO_RENDER_API SOFTWARE_FAST; ~/.local/ABDownloadManager/bin/ABDownloadManager

**Comment by amir1376** [comment]
Please try to compile the master branch to  see if it fixes the problem for you (with the default renderApi) ?
